libshortfin/src/shortfin/local/messaging.h

```diff
@@ -189,10 +189,11 @@ class SHORTFIN_API Queue : public std::enable_shared_from_this<Queue> {
   void Close();
 
  protected:
+  Queue(Options options);
+
  private:
```

Comment:
If we want to keep this constructor private, we could also rework the anonymous namespace used for `QueueCreator`, which is already tagged `friend struct QueueCreator`.

Reply:
* https://stackoverflow.com/questions/19872920/is-it-possible-to-friend-a-class-in-an-unnamed-namespace-in-c
* https://stackoverflow.com/questions/63397661/gcc-doesnt-like-to-make-friends-with-anonymous-namespace-forward-declarations

Reply:
Found a different fix: https://stackoverflow.com/a/63398702

Apparently `friend struct QueueCreator` was an "elaborated type specifier", and that is not needed here since the name is not ambiguous. gcc is happier with just `friend QueueCreator`.